openedx/openedx-platform
# Learner Dashboard improvements to assist with the Account Activation process

This change adds two optional features designed to ease learner confusion through the account activation and enrollment process.

Currently, when learners haven't yet activated their accounts, the Learner Dashboard shows both a warning about not being activated and the text below about not being enrolled in any courses. Many of our client's students don't read the first warning, get confused, and then email them about not being enrolled in the right courses.

As part of the upstream review process (see https://github.com/edx/edx-platform/pull/15464#issuecomment-362029099), we've also deprecated the top (black) banner, removed the activation message from the top banner, and permanently enabled the the sidebar activation message.

**JIRA tickets**: [OSPR-1814](https://openedx.atlassian.net/browse/OSPR-1814)

**Screenshots**:

There are three potential stages to this process, and this change improves two of them:

1. Learner has just registered, and has not yet activated their account, and is viewing the Dashboard page.
  Here is the confusing case, where the activation message in the sidebar is being missed by learners, who skip instead to the "You are not enrolled in any courses yet." message.
![show dashboard](https://user-images.githubusercontent.com/7556571/38247314-fdd12516-3783-11e8-96da-ebb7a49ac422.png)

  This change adds a Site Configuration and `FEATURE` flag, `HIDE_DASHBOARD_COURSES_UNTIL_ACTIVATED`, which hides the "My Courses" and "You are not enrolled in any courses" area.  This will ease learner confusion by making the activation message the only text shown on the page.
  Note that even if the learner was enrolled in a course prior to account activation, the system doesn't allow the user to visit the course without first activating their account.
![hide_dashboard_until_activated](https://user-images.githubusercontent.com/7556571/38247329-0ac83232-3784-11e8-8096-cd56d63c0ca5.png)

1. Learner has registered, has not yet activated their account, but has logged out or let their session expire. Learner has to login again, and/or reset their password.
  After login, the system re-sends the activation email, and shows an appropriate message, and won't let the learner proceed past the login screen until the account is activated.
  No change required here; the learner's next step is clear.
1. Learner has activated their account, is logged in, and viewing an empty dashboard with no courses in it.
  Here, we've added some optional text, which can be set in Site Configuration `EMPTY_DASHBOARD_MESSAGE`.  Our client had some specific information to convey at this point, and so configurable markup was appropriate.
![empty dashboard custom message](https://user-images.githubusercontent.com/7556571/37694913-e62de098-2d1a-11e8-89cb-3dd15b67a61b.png)

**Sandbox URL**:

https://pr15464.sandbox.opencraft.hosting (3357a25)

**Partner information**: 3rd party-hosted open edX instance

**Deployment targets**: edx.org and edge.edx.org

**Merge deadline**: ASAP

**Testing instructions**:

The `staff` user has superuser privileges on the sandbox server.
*Note* You must wait **3 min** for any Site Configuration settings changes to take effect.

1. Ensure the Site Configuration doesn't have `"HIDE_DASHBOARD_COURSES_UNTIL_ACTIVATED"`, or sets it to: `false` (default).
1. Register a new user, but don't follow the activation link yet.
   Note the "activate your account" message is visible in the Sidebar, and the top banner is not shown.
1. Update the Site Configuration to set `"HIDE_DASHBOARD_COURSES_UNTIL_ACTIVATED": true` and refresh the Dashboard.
   Note that the "My Courses" list is now hidden, and the sidebar is shifted left.
1. Visit the activation email link.
   Note that your account has been activated, and the "My Courses" list is now shown.
1. Update the Site Configuration to add an `EMPTY_DASHBOARD_MESSAGE` like the one shown in Site Configuration below.
1. Refresh the Dashboard. 
   Note that  the custom empty dashboard message is now displayed.
1. Enrol in a course, and revisit the Dashboard. 
   Note that the empty dashboard messages are no longer displayed.

**Reviewers**
- [ ] @marcotuts 

**Site Configuration**
```json
{
  "HIDE_DASHBOARD_COURSES_UNTIL_ACTIVATED": true,
  "EMPTY_DASHBOARD_MESSAGE": "If you believe this is an error, please contact <a href='mailto:contact@example.com'>contact@example.com</a>.<br /> If you would like to enroll in one of our free courses, you can find them in the <a href='/courses'>course catalog</a>.",
  "ACTIVATION_EMAIL_SUPPORT_LINK": "mailto:contact@example.com",
}
